test_cases/stdlib/check_sqlite3.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+import sqlite3
+from contextlib import closing
+from typing_extensions import assert_type
+
+# https://github.com/python/typeshed/issues/11558
+
+
+class MyConnection(sqlite3.Connection):
+    pass
+
+
+# Default return-type is Connection.
+with closing(sqlite3.connect(":memory:")) as con1:
+    assert_type(con1, sqlite3.Connection)
+
+# Providing an alternate factory changes the return-type.
+with closing(sqlite3.connect(":memory:", factory=MyConnection)) as con2:
+    assert_type(con2, MyConnection)
+
+# Provides a true positive error. When checking the connect() function,
+# mypy should report an arg-type error for the factory argument.
+try:
+    with closing(sqlite3.connect(":memory:", factory=None)) as con3:  # type: ignore
+        pass
+except TypeError:
+    pass  # Expected behavior.
+else:
+    raise RuntimeError("Expected TypeError, factory is not optional.")
```

Comment:
Since we don't run the tests, just run the type checker against them, this can be simplified to:

```suggestion
with sqlite3.connect(":memory:", factory=None) as con3:  # type: ignore
    pass
```